### ISSUE TITLE
No order_id in ps_mollie table

### ISSUE THREAD

**Issue by DjPaul78** [mollie/PrestaShop1.7#566]
Since last week I experience some problems. Because my site is 'born' from V1.4 but the current version is 1.7.6.0 for a while.
I noticed that the cart ID sometimes already existed in the order table, so I altered the ps_cart table and set the auto_increment to the highest value in the ps_orders table (+1).
After that, the table ps_mollie doesn't recieve any order_id anymore (NULL) and also the updated_at is NULL. But I think the updated_at field is not a big problem.
I already updated the mollie module, and did a reset. Can you perhaps tell me which file and function looks for the order_id so I can look into this? Or maybe you had this problem before and have some hints so I can look into that?
The orders are placed correctly but I created my own software which handles almost everything. So if a order s placed, it checks if the order is valid and payd. After that, it collects the products and sends it to the distributor, creates an invoice and sends an email to the customer with the invoice. But due to the empty field it can't find if the order is payd.

Thanks!

**Comment by justelis22** [comment]
Hello there, @DjPaul78!

Could you please clarify whether the issue happens for all of the orders or just some specific ones?
Also, could you whether the order made is finished when you see the NULL value, or is it just created and unpaid? As order_id is set when the order is created after payment is finished. If you create a payment but don't finish it then the order is not created and so there is no order_id.

This could be related to a faulty webhook, so it would be really helpful if you could check the order history on your account at mollie.com and see if the webhook call was successful, or if any error can be seen.

Waiting for your reply!

Best regards,
Invertus support team.

**Comment by DjPaul78** [comment]
That's the stranges thing. Every goes well, I see the payments in the mollie dashboard and in the bo, I see all new orders with state payment accepted. But in the ps_mollie_payments, the bank_status is set to open. 
I just found out that the PayPal module has almost the same problem. So I think the problem is coming from somewhere else.
Can you maybe tell me which file makes a request to get the order_id if an order is created or which class and function is called?
From there I can see which values I get back from de db.

Regards,
Paul

**Comment by justelis22** [comment]
Hey there, @DjPaul78!

I have forwarded your query to our development team, we will get back to you once we have an answer!

Best regards,
Invertus support team.

**Comment by DjPaul78** [comment]
Never mind, I found the problem. I had added a variable inside the classes/PaymentModule. From there I added a new file to the folder override/classes and added an override PaymentModule with: 
abstract class PaymentModule extends PaymentModuleCore
{
    public function validateOrder($id_cart etc...)
   {
    //Line 575 added '{id_order}' => $order->id,
   }
}

But somehow that doesn't work. So I deleted the whole PaymentModule.php from the override and added the var in the original file. Not very nice but... It works now!

Thanks you very very much for your great support!
Paul
